packages/react-components/src/components/data-providers/tsp-data-provider.ts

```diff
@@ -48,106 +49,113 @@ export class TspDataProvider {
+            const start = totalTimeRange.getStart() + viewRange.start;
+            const end = totalTimeRange.getStart() + viewRange.end;
+            const timeGraphStateParams = QueryHelper.selectionTimeRangeQuery(start, end, nbTimes, ids);
+            const statesPromise = this.client.fetchTimeGraphStates(this.traceUUID, this.outputId, timeGraphStateParams);
 
-        if (annotationMarkers) {
-            additionalProps['requested_marker_categories'] = annotationMarkers;
-        }
-        if (markerSetId) {
-            additionalProps['requested_marker_set'] = markerSetId;
-        }
-
-        const annotationParams = QueryHelper.selectionTimeRangeQuery(start, end, nbTimes, ids, additionalProps);
+            const additionalProps: { [key: string]: any } = {};
+            if (annotationMarkers) {
+                additionalProps['requested_marker_categories'] = annotationMarkers;
+            }
+            if (markerSetId) {
+                additionalProps['requested_marker_set'] = markerSetId;
+            }
+            const annotationParams = QueryHelper.selectionTimeRangeQuery(start, end, nbTimes, ids, additionalProps);
+            const annotations: Map<number, TimelineChart.TimeGraphAnnotation[]> = new Map();
+            const annotationsPromise = this.client.fetchAnnotations(this.traceUUID, this.outputId, annotationParams);
 
-        const annotations: Map<number, TimelineChart.TimeGraphAnnotation[]> = new Map();
-        const tspClientResponse2 = await this.client.fetchAnnotations(this.traceUUID, this.outputId, annotationParams);
-        const annotationsResponse = tspClientResponse2.getModel();
-        const rangeEvents: TimelineChart.TimeGraphAnnotation[] = [];
+            const arrowStart = viewRange.start + this.timeGraphEntries[0].start;
+            const arrowEnd = viewRange.end + this.timeGraphEntries[0].start;
+            const fetchParameters = QueryHelper.timeRangeQuery(arrowStart, arrowEnd, nbTimes);
+            const arrowsPromise = this.client.fetchTimeGraphArrows(this.traceUUID, this.outputId, fetchParameters);
 
-        if (tspClientResponse2.isOk() && annotationsResponse) {
-            Object.entries(annotationsResponse.model.annotations).forEach(([category, categoryArray]) => {
-                categoryArray.forEach(annotation => {
-                    if (annotation.type === Type.CHART) {
-                        if (annotation.entryId === -1) {
-                            rangeEvents.push(this.getAnnotation(category, annotation, rangeEvents.length, chartStart));
-                        } else {
-                            let entryArray = annotations.get(annotation.entryId);
-                            if (entryArray === undefined) {
-                                entryArray = [];
-                                annotations.set(annotation.entryId, entryArray);
+            const tspClientAnnotationsResponse = await annotationsPromise;
```

Comment:
when executing this await, all 3 created promises (for annotations, states and arrows) are triggered and the back-end requests are done. With this all 3 are sent in parallel. However, we should make it more explicit and schedule all 3 executions using Promise.all() or Promise.allSettled().